### ISSUE TITLE
Search: WP Query tax searches with AND/OR probably not working

### ISSUE THREAD

**Issue by gibrown** [Automattic/jetpack#8286]
Mentioned by @nickdaugherty in #8276 and opening as a separate issue.

cc @gravityrail

**Comment by gravityrail** [comment]
Ping @nickdaugherty - can you provide any clarity on the exact issue here and how to reproduce it?

**Comment by gravityrail** [comment]
The issue is as follows:

- WP lets you specify AND or OR as operators on a taxonomy term query
- Our [current implementation](https://github.com/Automattic/jetpack/blob/master/modules/search/class.jetpack-search.php#L503-L535) assumes "AND"
- We need to adapt our current implementation to detect and use OR if specified.

**Comment by nickdaugherty** [comment]
Yep that's the gist of it. I also realized we don't consider the `operator` of a Tax Query in our implementation, which can be:

> Possible values are ‘IN’, ‘NOT IN’, ‘AND’, ‘EXISTS’ and ‘NOT EXISTS’. Default value is ‘IN’.

**Comment by Viper007Bond** [comment]
This should only affect people who are modifying the search results via `pre_get_posts` I think so going to punt this for now. Edge case and we're running out of time.
